### ISSUE TITLE
Javascript error on invalid Elm code

### ISSUE THREAD

**Issue by emmanueltouzery** [elm/compiler#168]
When trying to write my first Elm code, elm didn't give compile errors but a javascript error at runtime on this code (which I'm pretty sure is bogus):

``` haskell
import Http

serverJson : Signal String
serverJson = Http.sendGet (constant "http://echo.jsontest.com/key/value")

showResults : String -> Element
showResults json = flow down [plainText "Hour" , plainText json, plainText "Description", plainText "Extra info"]

main = lift showResults serverJson
```

The error is:

```
TypeError: xs is undefined  
while (xs.ctor !== 'Nil') {
elm-0.8.js line 649.
```

elm 0.8.0.3.
